UpdateProfile/__tests__/handler.test.ts

```diff
@@ -109,6 +115,277 @@ describe("UpdateProfileHandler", () => {
+    };
+
+    const updateProfileHandler = UpdateProfileHandler(profileModelMock as any);
+
+    const result = await updateProfileHandler(contextMock as any, aFiscalCode, {
+      ...aProfile
+    });
+
+    expect(result.kind).toBe("IResponseSuccessJson");
+    if (result.kind === "IResponseSuccessJson") {
+      expect(result.value).toEqual(
+        expect.objectContaining({
+          service_preferences_settings: {
+            mode: legacyProfileServicePreferencesSettings.mode,
+            version: legacyProfileServicePreferencesSettings.version
+          }
+        })
+      );
+    }
+  });
+
+  it("should not increment service_preferences_settings.version if mode remains AUTO", async () => {
+    const profileModelMock = {
+      findLastVersionByModelId: jest.fn(() =>
+        // Return a profile with a validated email
+        taskEither.of(some({ ...aRetrievedProfile, servicePreferencesSettings: autoProfileServicePreferencesSettings }))
+      ),
+      update: jest.fn(_ => taskEither.of({ ...aRetrievedProfile, ..._ }))
+    };
+
+    const updateProfileHandler = UpdateProfileHandler(profileModelMock as any);
+
+    const result = await updateProfileHandler(contextMock as any, aFiscalCode, {
+      ...aProfile,
+      service_preferences_settings: autoApiProfileServicePreferencesSettings
+    });
+
+    expect(result.kind).toBe("IResponseSuccessJson");
+    if (result.kind === "IResponseSuccessJson") {
+      expect(result.value).toEqual(
+        expect.objectContaining({
+          service_preferences_settings: {
+            mode: autoProfileServicePreferencesSettings.mode,
+            version: autoProfileServicePreferencesSettings.version
+          }
+        })
+      );
+    }
+  });
+
+  it("should not increment service_preferences_settings.version if mode remains MANUAL", async () => {
+    const profileModelMock = {
+      findLastVersionByModelId: jest.fn(() =>
+        // Return a profile with a validated email
+        taskEither.of(some({ ...aRetrievedProfile, servicePreferencesSettings: manualProfileServicePreferencesSettings }))
+      ),
+      update: jest.fn(_ => taskEither.of({ ...aRetrievedProfile, ..._ }))
+    };
+
+    const updateProfileHandler = UpdateProfileHandler(profileModelMock as any);
+
+    const result = await updateProfileHandler(contextMock as any, aFiscalCode, {
+      ...aProfile,
+      service_preferences_settings: manualApiProfileServicePreferencesSettings
+    });
+
+    expect(result.kind).toBe("IResponseSuccessJson");
+    if (result.kind === "IResponseSuccessJson") {
+      expect(result.value).toEqual(
+        expect.objectContaining({
+          service_preferences_settings: {
+            mode: manualProfileServicePreferencesSettings.mode,
+            version: manualProfileServicePreferencesSettings.version
+          }
+        })
+      );
+    }
+  });
+
+  it("should increment service_preferences_settings.version if mode has changed from LEGACY to MANUAL", async () => {
```

Comment:
is missing the test case where we change the mode from LEGACY to AUTO.

Reply:
fixed in 01360d9